src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java

```diff
@@ -240,6 +258,53 @@
         });
     }
 
+    @Issue("JENKINS-30326")
+    @Test
+    public void testGlobalBindingWithAuthorization() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                // configure security
+                story.j.jenkins.setSecurityRealm(story.j.createDummySecurityRealm());
+                story.j.jenkins.setAuthorizationStrategy(new FullControlOnceLoggedInAuthorizationStrategy());
+                // create the user.
+                story.j.jenkins.getUser("dummy");
+                
+                // enable the run as user strategy for the AuthorizeProject plugin
+                Map<String, Boolean> strategies = new HashMap<String, Boolean>();
+                strategies.put(story.j.jenkins.getDescriptor(SpecificUsersAuthorizationStrategy.class).getId(), true);
+                QueueItemAuthenticatorConfiguration.get().getAuthenticators().add(new ProjectQueueItemAuthenticator(strategies));
```

Comment:
any change would be caught by the assertion on line 303  `assertLogContains("running as user: dummy", b);`
FWIW this is how the tests in AuthorizeProject are written.